src/services/createRouter.ts

```diff
@@ -47,7 +48,10 @@ type RouterUpdateOptions = {
  * const router = createRouter(routes)
  * ```
  */
-export function createRouter<const T extends Routes>(routes: T, options: RouterOptions = {}): Router<T> {
+export function createRouter<const T extends Routes>(routes: T, options?: RouterOptions): Router<T>
+export function createRouter<const T extends Routes>(arrayOfRoutes: T[], options?: RouterOptions): Router<T>
```

Comment:
I'm surprised we don't need to flatten this. But I suppose T is a union so we get a union of routes on the router?

Reply:
I agree, I was pleasantly surprised but it definitely does get correctly merged types

<img width="641" alt="image" src="https://github.com/kitbagjs/router/assets/6098901/2824f9bf-7f42-4a61-9fe3-a04892a35622">

Reply:
Nice, simpler than I imagined.